test/e2e/recreate_deployment_test.go

```diff
@@ -54,3 +56,57 @@ func TestRecreateDeployment(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestRestoreDeploymentAfterUserChanges(t *testing.T) {
+	client := framework.MustNewClientset(t, nil)
+
+	defer framework.MustRemoveImageRegistry(t, client)
+	framework.MustDeployImageRegistry(t, client, nil)
+	framework.MustEnsureImageRegistryIsAvailable(t, client)
+
+	// add a new environment variable and a host port to the deployment.
+	if _, err := client.Deployments(framework.OperatorDeploymentNamespace).Patch(
+		defaults.ImageRegistryName,
+		types.JSONPatchType,
+		[]byte(`[
+			{
+				"op": "add",
+				"path": "/spec/template/spec/containers/0/env/-",
+				"value": {"name": "FOO", "value": "BAR"}
+			},
+			{
+				"op": "add",
+				"path": "/spec/template/spec/containers/0/ports/-",
+				"value": {"name": "foo", "containerPort": 2222}
+			}
+		]`),
+	); err != nil {
+		t.Fatalf("unable to patch image registry deployment: %v", err)
+	}
+
+	// give some room for the operator to restore the deployment.
+	time.Sleep(10 * time.Second)
```

Comment:
It's better to watch the resource. Or at least to poll. You never know how slow the CI machine can be.

Reply:
I added a Poll here, PTAL.